### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by shingoed**

### CODE REVIEW

**Comment by shingoed** [comment]
- [ ] Top-level README “Table of Contents” is updated
 - [ ] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [ ] README for this challenge is complete
     - [ ] Summary, Description, Approach & Efficiency, Solution
     - [ ] Link to code
     - [ ] Picture of whiteboard

File: code401challenges/src/test/java/code401challenges/LibraryTest.java [hunk 1]
```diff
@@ -11,20 +11,42 @@
 import static org.junit.Assert.*;
 
 public class LibraryTest {
-    @Test public void testSomeLibraryMethod() {
-        int[] testArrayEvenLength = new int[] {1,2,3,4,10,11};
-        int testValue1 = 99;
-        int[] expectedResult = new int[] {1,2,3,99,4,10,11};
 
-        assertTrue(Arrays.equals(ArrayShift.shiftArray(testArrayEvenLength, testValue1), expectedResult));
+    @Test
```
Comment (SharinaS) [inline code401challenges/src/test/java/code401challenges/LibraryTest.java]: When writing a test for a class, you need to make a new test class for each class. So, your test class for Binary Search should have a `public class BinarySearchTest`.

File: code401challenges/src/main/java/code401challenges/BinarySearch.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package code401challenges;
+
+import java.util.Arrays;
+//resource: https://www.baeldung.com/java-binary-searchhttps://www.baeldung.com/java-binary-search
+
+public class BinarySearch {
+
+    public static int binarySearch(int[] sortedArray, int key) {
+        int left = 0;
+        int right = sortedArray.length;
```
Comment (SharinaS) [inline code401challenges/src/main/java/code401challenges/BinarySearch.java]: Try: `int right = sortedArray.length-1;`
